Assignment 5.Rmd

```diff
@@ -60,18 +72,34 @@ plot(score_ctree)
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
+Teachers should pay attention to hint and prior_prob_count. Students who has asked for hints are generally more likely to 
+
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("intelligent_tutor_new.csv")
 
 #Generate predicted advice for new students based on tree generated from old students
 
 D2$prediction <- predict(score_ctree, D2)
-
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+```{r}
+D1$advice <- as.factor(D1$advice)
+table(D1$score)
+D2$advice <- ifelse(D2$score <=0.4, "intervene", ifelse(D2$score > 0.4 & D2$score <=0.8, "monitor", "no action"))
+plot(D1$advice)
+plot(D2$prediction)
+
+# calculate the accuracy 
+D2$correct <- ifelse(D2$prediction == D2$advice, 1, 0)
+tab <- table(D2$correct)
+acc <- tab[2] / (tab[1] + tab[2]) 
+acc <- as.numeric(acc)
+acc
```

Comment:
What is your conclusion from this analysis? You could also run the same analysis directly from the data frame: `mean(ifelse(D2$prediction == "no action", 1, 0))`